tasks/main.yml

```diff
@@ -1,47 +1,8 @@
-# Copyright 2014 Google Inc. All rights reserved.
-#
-# Licensed under the Apache License, Version 2.0 (the "License");
-# you may not use this file except in compliance with the License.
-# You may obtain a copy of the License at
-#
-#     http://www.apache.org/licenses/LICENSE-2.0
-#
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
 ---
-- name: Check that the Stackdriver API key has been specified.
-  fail: msg="Please provide your Stackdriver API key in the variable stackdriver_api_key."
-  when: stackdriver_api_key is not defined or not stackdriver_api_key
+- name: Fail if agent type is not supported
+  fail:
+    msg: "The Cloud Ops Ansible role only supports the following agents: \"monitoring\", \"logging\" and \"ops_agent\"."
+  when: agent_type not in ["monitoring"]
```

Comment:
Structuring this as a fail/when will cause this to show up as a "skipped" task in the normal case. I would instead refactor this to use the [assert](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/assert_module.html) module, which should show up as a success.